assembl/static2/js/app/utils/extract.jsx

```diff
@@ -125,25 +129,49 @@ export const ActionIcons = ({ qualifier, backgroundColor, color }) => {
   }
 };
 
-export const NatureIcons = ({ qualifier }) => {
-  switch (qualifier) {
-  case 'concept':
-    return <BlueFlag />;
-  case 'argument':
-    return <YellowFlag />;
-  case 'example':
-    return <OrangeFlag />;
-  case 'issue':
-    return <RedFlag />;
-  case 'actionable_solution':
-    return <GreenFlag />;
-  case 'knowledge':
-    return <PurpleFlag />;
-  case 'cognitive_bias':
-    return <BlackFlag />;
-  default:
-    return <span>{qualifier}</span>;
+function getTaxonomyNatureColor(nature: string): string {
+  const mapping = {
+    concept: '#00B6FF',
+    argument: '#FFEC00',
+    example: '#FF9F00',
+    issue: '#FF001F',
+    actionable_solution: '#35C646',
+    knowledge: '#BD10E0',
+    cognitive_bias: '#000000'
+  };
+
+  return get(mapping, nature, null);
+}
+
+type NatureIconProps = {
+  qualifier: string
+};
+
+export const NatureIcons = ({ qualifier }: NatureIconProps) => {
+  const color = getTaxonomyNatureColor(qualifier);
+  if (color) {
+    return <Flag color={color} />;
   }
+
+  return <span>{qualifier}</span>;
 };
 
-export const getExtractTagId = id => `message-body-local:Content/${id}`;
\ No newline at end of file
+export const getExtractTagId = (id: number) => `message-body-local:Content/${id}`;
+
+export function getExtractColor(nature: string, state: ExtractState, extractedByMachine: boolean): string {
+  const defaultColor = '#7ed321';
```

Comment:
@cedricmessiant same remark as above about using explicit constant name for a color